File: app/src/main/java/seveida/firetvforreddit/response/objects/Resolution.java
package seveida.firetvforreddit.response.objects;

import com.squareup.moshi.Json;

public class Resolution {

    @Json(name = "url")
    public String url;
    @Json(name = "width")
    public int width;
    @Json(name = "height")
    public int height;

    /**
     * No args constructor for use in serialization
     *
     */
    public Resolution() {
    }

    /**
     *
     * @param height
     * @param width
     * @param url
     */
    public Resolution(String url, int width, int height) {
        super();
        this.url = url;
        this.width = width;
        this.height = height;
    }

}

File: response_objects/src/main/java/baron/severin/response_objects/raw/response/api/json/Resolution.kt
package baron.severin.response_objects.raw.response.api.json

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Resolution(
    @Json(name = "url") val url: String,
    @Json(name = "width") val width: Int = 0,
    @Json(name = "height") val height: Int = 0
)
